### PR TITLE
pam_krb5 migration: import document

### PR DESCRIPTION
**Pull request by pbrezina**
The document was reformatted.

### CODE REVIEW

File: src/docs/pam_krb5_migration.rst [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Migrating from pam_krb5
+=======================
+
+``pam_krb5`` was a Pluggable Authentication Module (PAM) for performing user
+session authentication against Kerberos (specifically krb5). Red Hat formally
+announced its deprecation in the RHEL-7.4 release.
+
+System Security Services Daemon (SSSD) is a broader toolsuite for managing
+authentication mechanisms and remote directories. It includes a PAM module,
+``pam_sss``, which can perform the tasks for which ``pam_krb5`` was previously
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:11]: s/for which/where/

File: src/docs/pam_krb5_migration.rst [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+used.
+
+Architecturally, ``pam_krb5`` was a monolithic module which performed all needed
+tasks within itself. SSSD is set up differently: the module ``pam_sss`` calls
+out to the running SSSD itself for most functionality.
+
+.. note::
+    SSSD does not currently handle the AFS capabilitqies of ``pam_krb5``. If you
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:19]: s/capabilitqies/capabilities/

File: src/docs/pam_krb5_migration.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    would like such functionality, please contact Red Hat Support.
+
+Core functionality
+------------------
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:23]: Perhaps this section should be called ``Configuration`` instead?

File: src/docs/pam_krb5_migration.rst [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+The ``pam_krb5`` ``ccache_dir``, ``ccname_template``, ``keytab``, and
+``validate`` or ``no_validate`` options map to the ``SSSD.conf`` options
+``krb5_ccachedir``, ``krb5_ccname_template``, ``krb5_keytab``, and
+``krb5_validate``, respectively. By default, validation is not enabled, unless
+the Kerberos provider is IPA or Active Directory. See ``man SSSD-krb5(5)`` for
+more information.
+
+``pam_krb5``'s ``validate_user_user``, ``multiple_ccaches``, and
+``cred_session`` options have no SSSD equivalent. Please contact Red Hat Support
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:130]: s/validate_user_user/validate_user/ ?

File: src/docs/pam_krb5_migration.rst [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+SSSD refuses to admit users that do not exist (i.e., that cannot be resolved
+through its NSS interface). Therefore, the ``no_user_check`` option from
+``pam_krb5`` (previously marked as potentially dangerous) has no SSSD analogue.
+
+Other
+^^^^^
+
+``pam_krb5``'s ``ignore_unknown_principals``, ``ignore_unknown_spn``, and
+``ignore_unknown_upn`` options are united as the ``pam_sss``
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:156]: s/united/combined/

File: src/docs/pam_krb5_migration.rst [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+The PAM configuration for SSSD is very similar:
+
+.. code-block:: pam
+
+    auth required pam_sss.so
+    session optional pam_sss.so
+    account sufficient pam_sss.so
+    password sufficient pam_sss.so
+
+and in SSSD.conf we would add:
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:61]: s/SSSD.conf/sssd.conf/ ?

File: src/docs/pam_krb5_migration.rst [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+for AFS: ``afs_cells``, ``external``, ``ignore_afs``, ``null_afs``, ``tokens``,
+``tokens_strategy``.
+
+Debugging
+^^^^^^^^^
+
+Debugging in ``pam_krb5`` was controlled by the ``debug``, ``debug_sensitive``,
+and ``trace`` options. The debugging of ``pam_sss`` is not configurable
+(everything is logged and can be filtered appropriately). Debugging SSSD itself
+is controlled by the ``debug_level`` parameter of ``SSSD.conf``, and can also be
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:94]: s/SSSD.conf/sssd.conf/ ?

File: src/docs/pam_krb5_migration.rst [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+Prompting
+^^^^^^^^^
+
+``pam_krb5`` used the options ``banner``, ``chpw_prompt``, and ``pwhelp`` to
+display custom messages to the user when prompting for input. There is no direct
+replacement for these; however, an expiration messsage
+(``pam_account_expired_message``) and account lockout message
+(``pam_account_locked_message``) can be set. See documentation on
+``pam_verbosity`` (in ``SSSD.conf(5)``) for more information on when these are
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:108]: s/SSSD.conf/sssd.conf/ ?

File: src/docs/pam_krb5_migration.rst [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+controls in ``pam_sss``. In particular, ``use_authtok`` and ``use_first_pass``
+keep name and functionality. The behavior coded by ``try_first_pass`` is the
+SSSD default. All other prompting is can be enabled by setting
+``prompt_always``. For more information, see ``pam_sss(8)``.
+
+Credential management
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``pam_krb5`` ``ccache_dir``, ``ccname_template``, ``keytab``, and
+``validate`` or ``no_validate`` options map to the ``SSSD.conf`` options
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:123]: s/SSSD.conf/sssd.conf/ ?

File: src/docs/pam_krb5_migration.rst [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+``krb5_ccachedir``, ``krb5_ccname_template``, ``krb5_keytab``, and
+``krb5_validate``, respectively. By default, validation is not enabled, unless
+the Kerberos provider is IPA or Active Directory. See ``man SSSD-krb5(5)`` for
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:126]: s/man SSSD-krb5(5)/man sssd-krb5(5)/ ?

File: src/docs/pam_krb5_migration.rst [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+``pam_krb5``'s ``validate_user_user``, ``multiple_ccaches``, and
+``cred_session`` options have no SSSD equivalent. Please contact Red Hat Support
+if you would like such functionality.
+
+Localauth / .k5login
+^^^^^^^^^^^^^^^^^^^^
+
+SSSD and ``pam_krb5`` have different approaches here. In order to enable
+``.k5login``-based access control, set ``access_provider`` to krb5 in
+``SSSD.conf``. SSSD also includes its own localauth plugin (which is typically
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:138]: s/SSSD.conf/sssd.conf/ ?

File: src/docs/pam_krb5_migration.rst [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+
+``pam_krb5``'s ``ignore_unknown_principals``, ``ignore_unknown_spn``, and
+``ignore_unknown_upn`` options are united as the ``pam_sss``
+``ignore_unknown_user`` option. However, as above, note that SSSD will not
+authenticate users that cannot be resolved. Additionally, for users whose
+information cannot be obtained from LDAP (or who do not exist in LDAP), SSSD
+falls back to ``username@REALM``.
+
+``minimum_uid`` is called ``min_id`` in ``SSSD.conf``. Additionally,
+``SSSD.conf`` allows ``max_id`` to limit the maximum UID to check. Note that
```
Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:163]: s/SSSD.conf/sssd.conf/ ?

Comment (justin-stephenson) [inline src/docs/pam_krb5_migration.rst:163]: s/SSSD.conf/sssd.conf/ ?

File: src/docs/pam_krb5_migration.rst (hunk 3 above)
Comment (pbrezina) [reply src/docs/pam_krb5_migration.rst:23]: Agree.

**Comment by pbrezina** [comment]
Thank you. New version is available.

**Comment by justin-stephenson** [comment]
LGTM

**Pull request merged**